src/App.php

```diff
@@ -565,8 +566,9 @@ public function getCurrentThemeStylesheetPath(): string
 	 */
 	public function runFrontend(App\Router $router, IManagePersonalConfigValues $pconfig, Authentication $auth, App\Page $page, Nav $nav, ModuleHTTPException $httpException, HTTPInputData $httpInput, float $start_time, array $server)
 	{
-		$requeststring = ($_SERVER['REQUEST_METHOD'] ?? '') . ' ' . ($_SERVER['REQUEST_URI'] ?? '') . ' ' . ($_SERVER['SERVER_PROTOCOL'] ?? '');
-		$this->logger->debug('Request received', ['address' => $_SERVER['REMOTE_ADDR'] ?? '', 'request' => $requeststring, 'referer' => $_SERVER['HTTP_REFERER'] ?? '', 'user-agent' => $_SERVER['HTTP_USER_AGENT'] ?? '']);
+		DID::routeRequest($this->args->getCommand(), $server);
```

Comment:
You can move this call inside of the `try` block further down and this allows you to make `DID::notFound()` just throw a `HTTPException\NotFoundException` and it will be picked up by the regular error handling.

Reply:
Done